### PR TITLE
kv,util: check TiKV version before event feed and refine version check

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Check TiKV version before making EventFeed RPC in order to mitigate compatibility issues.

Cc  #663 

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

 - Possible performance regression
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

- No release note.

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -291,13 +291,18 @@ func (c *CDCClient) getConn(ctx context.Context, addr string) (*grpc.ClientConn,
...
 	err = retry.Run(50*time.Millisecond, 3, func() error {
 		conn, err := c.getConn(ctx, addr)
 		if err != nil {
 			log.Info("get connection to store failed, retry later", zap.String("addr", addr), zap.Error(err))
 			return errors.Trace(err)
 		}
+		err = util.CheckStoreVersion(ctx, c.pd, storeID)
+		if err != nil {
+			log.Error("check tikv version failed", zap.Error(err), zap.Uint64("storeID", storeID))
+			return errors.Trace(err)
```
Comment (amyangfei) [inline cdc/kv/client.go:304]: can use `backoff.Permanent(err)` here to avoid unnecessary retries

**Comment by overvenus** [comment]
> Firstly we perform rolling update of cdc cluster, but the CheckClusterVersion (which requires tikv-v2) in cdc server will fail the startup of cdc-v2 server.

There is a workaround to break the tie,

* Relaxed version check on startup, warn if TiKV is incompatible, error if PD is incompatible.
* And sends RPC `EventFeed` util TiKV is upgraded.

File: cdc/kv/client.go [hunk 2]
```diff
@@ -621,14 +626,19 @@ MainLoop:
...
-					zap.Uint64("regionID", sri.verID.GetID()), zap.Uint64("requestID", requestID), zap.String("addr", rpcCtx.Addr))
-				stream, err = s.client.newStream(ctx, rpcCtx.Addr)
+					zap.Uint64("regionID", sri.verID.GetID()),
+					zap.Uint64("requestID", requestID),
+					zap.Uint64("storeID", storeID),
+					zap.String("addr", rpcCtx.Addr))
+				stream, err = s.client.newStream(ctx, rpcCtx.Addr, storeID)
 				if err != nil {
 					// if get stream failed, maybe the store is down permanently, we should try to relocate the active store
 					log.Warn("get grpc stream client failed",
```
Comment (amyangfei) [inline cdc/kv/client.go:638]: A lot of `version is incompatible` errors are logged here, is it better to add some sleep for this error?

Comment (overvenus) [reply cdc/kv/client.go:638]: Updated.

**Pull request merged**